src/serialize.rs

```diff
@@ -123,6 +123,32 @@ where
     Ok((map, Default::default()))
 }
 
+/// Serialize for an Array. Required to hide the internal attributes.
+pub fn serialize_array<S>(
+    terms: &Vec<RichTerm>,
+    _attrs: &ArrayAttrs,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let mut seq = serializer.serialize_seq(Some(terms.len()))?;
+    for term in terms.iter() {
+        seq.serialize_element(term)?;
+    }
+
+    seq.end()
+}
+
+/// Deserialize for an Array. Required to set the default attributes.
+pub fn deserialize_array<'de, D>(deserializer: D) -> Result<(Vec<RichTerm>, ArrayAttrs), D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let terms: Vec<RichTerm> = Vec::deserialize(deserializer)?;
+    Ok((terms, Default::default()))
+}
+
```

Comment:
Hm. It seems array serialization tests are failing, because of an extra `[]` added by serde.

So instead of `{ "bar": ["str", true] }` w get `{ "bar": [ ["str", true] ] }`. Maybe this happened because we now have a two-element tuple instead of one vector. I'll look into serde attributes.